test/common/event/guarddog_impl_test.cc

```diff
@@ -0,0 +1,155 @@
+#include "common/common/utility.h"
+
+#include "server/guarddog_impl.h"
+
+#include "test/mocks/common.h"
+#include "test/mocks/server/mocks.h"
+#include "test/mocks/stats/mocks.h"
+
+using testing::InSequence;
+
+namespace Server {
+
+/**
+ * Death test caveat: Because of the way we die gcov doesn't receive coverage
+ * information from the forked process that is checked for succesful death.
+ * This means that the lines dealing with the calls to PANIC are not seen as
+ * green in the coverage report. However, rest assured from the results of the
+ * test: these lines are in fact covered.
+ */
+class GuardDogDeathTest : public testing::Test {
+protected:
+  GuardDogDeathTest()
+      : config_kill_(1000, 1000, 100, 1000), config_multikill_(1000, 1000, 1000, 500),
+        time_point_(std::chrono::system_clock::now()) {}
+
+  testing::NiceMock<Server::Configuration::MockMain> config_kill_;
```

Comment:
nit: Can we do using testing::NiceMock above and avoid the extra verbiage here and elsewhere in tests.